OSSMDOC-705: Update content  for Service Mesh Extension API deprecation (#51803)
openshift/openshift-docs: modules/ossm-extensions-migration-overview.adoc

[id="ossm-extensions-migration-overview_{context}"]
= Migrating from `ServiceMeshExtension` to `WasmPlugin` resources

> The `ServiceMeshExtension` API is deprecated as of {SMProductName} version 2.2 and was removed as of {SMProductName} version 2.3. If you are using the `ServiceMeshExtention` API, you must migrate to the `WasmPlugin` API to continue using your WebAssembly extensions.

Reviewer: ```suggestion
The `ServiceMeshExtension` API, which was deprecated in {SMProductName} version 2.2, is removed in {SMProductName} version 2.3. If you are using the `ServiceMeshExtention` API, you must migrate to the `WasmPlugin` API to continue using your WebAssembly extensions.
```
Trying to follow the present tense style we've used in https://redhat-documentation.github.io/supplementary-style-guide/#release-notes.

Reviewer: Reviewing the openshift-docs repo, there's a terminology split:
- 36 references to WebAssembly module
- 15 references to WebAssembly extension
